### DISCUSSION TITLE
Infer required/optional from validation schema

### DISCUSSION THREAD

**Discussion by MerlinMason** [react-hook-form/react-hook-form#10040] [Ideas]
If you want to consistently style required/optional form inputs differently you have to pass them a `required` prop to do so. Additionally you have to remember to ensure these props align with your validation schema (with opportunity for human error).

It would be really handy if you could determine whether a given field was optional or required by inferring this from the validation schema.

Potentially this could look something like:
```js
export const TextInputInfered = ({ name, label }: { name: string; label: string }) => {
    const { register, validation } = useFormContext() || {};

    return (
        <>
            <label htmlFor={name}>
                {label}
                {validation?.[name]?.required ? "" : "(optional)"}
            </label>

            <input {...register?.(name)} type="text" id={name} />
        </>
    );
};
```

Be interested to hear peoples thoughts on this... Is this already possible somehow? If not, would it be feasible to implement?

**Comment by eJayYoung** [comment]
good idea!
especially when control conditional form inputs, I should control required style display and validation schema in separate place.
so I intend if there is a way to define logic in one place would improve development experience efficiency.

**Comment by davidmarciano** [comment]
Totally agree, I was surprised when found out this is not supported by default.
